### ISSUE TITLE
VLoopers play back keyboard actions too fast

### ISSUE THREAD

**Issue by tonikone** [mdotstrange/NightmarePuppeteerPublic#179]
This is best replicated by starting a stopwatch and pressing single action keys at regular intervals, like beats of music, while counting the number of keypresses. Or initiate actions one by one and let each action loop a set amount of times.

**Comment by mdotstrange** [comment]
Yeah this is a known issue- I've tried several times to get it right but it always ends up off- has to do with Update timing- there's Update and Fixed Update running in normal scripts and Coroutines have their own update types- though I think the timing mismatch has to do with a floating point issue as I'm using an int variable as a frame counter then I convert to float to do some math to figure out the time to wait between frames-  I'll take a look at it again- thanks

**Comment by mdotstrange** [comment]
I moved the frame counter from Update to FixedUpdate- def seems slower now- didn't so a stopwatch test though- next build will have this change- can let me know if its more accurate then

**Comment by tonikone** [comment]
Do two beat single actions at 120 bpm and the result actions seem to be a dotted quarter note. So a cut of 25 percent per action. Full measure action times are also cut to about three quarters at 120 bpm. Maybe a rough area around 2/3 to 3/4 could give a clue. At least the errors seem to make some sort of rhythmic sense 😃

**Comment by mdotstrange** [comment]
I moved all the Vlooper code into FixedUpdate for the latest build on Steam now- hope it helps as I'm now out of bullets 0_0
